charts/daemonset-app/templates/_helpers.tpl

```diff
@@ -0,0 +1,16 @@
+{{/*
+This function generates an extended set of labels by combining the base labels 
+from the "nd-common.labels" template with additional custom labels. 
+
+The additional labels include:
+  - helm.chart/name: Specifies the name of the chart (hardcoded as "daemonset-app").
+  - helm.chart/version: Includes the chart version dynamically from .Chart.Version.
+*/}}
+{{- define "nd-common.extendedLabels" -}}
```

Comment:
This name should be something like `daemonset-app.extendedLabels` .... `nd-common` should only be used for functions inside the `charts/nd-common` chart.